php-frontend/src/main/java/org/sonar/php/metrics/CognitiveComplexityVisitor.java

```diff
@@ -66,6 +67,34 @@ public static CognitiveComplexity complexity(FunctionTree functionTree) {
     return cognitiveComplexityVisitor.complexity;
   }
 
+  public static int complexity(CompilationUnitTree cut) {
+    // Only visit functions, methods and function expressions.
+    // Nesting is taken into account by cognitive complexity (no need to visit recusively).
+    class CompilationUnitVisitor extends PHPVisitorCheck {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

Reply:
Ugly FP.